### PR TITLE
Added backup custom periodicity support

### PR DESCRIPTION
**Pull request by bastien70**
See #68 

All that remains is to modify the documentation.

### CODE REVIEW

File: migrations/Version20220801120509.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20220801120509 extends AbstractMigration
```
Comment (jmsche) [inline migrations/Version20220801120509.php:13]:
```suggestion
final class Version20220801120509 extends AbstractMigration
```

File: src/Entity/Embed/BackupTask.php [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+#[Embeddable]
+class BackupTask
+{
+    #[ORM\Column(type: Types::STRING, length: 255, enumType: BackupTaskPeriodicity::class)]
+    private ?BackupTaskPeriodicity $periodicity = null;
+
+    #[ORM\Column(type: Types::INTEGER)]
+    #[Assert\Type(type: 'integer')]
+    #[Assert\Count(min: 1)]
```
Comment (jmsche) [inline src/Entity/Embed/BackupTask.php:25]: IIRC count is for iterables

File: src/Service/BackupService.php [hunk 3]
```diff
@@ -154,6 +154,25 @@ public function getDatabases(): array
         return $this->databaseRepository->findAll();
     }
 
+    /**
+     * @return Database[]
+     */
+    public function getDatabasesToBackup(): array
```
Comment (jmsche) [inline src/Service/BackupService.php:160]: Can we do this at database level instead?

**Review by jmsche — CHANGES_REQUESTED** [review]

File: migrations/Version20220801120509.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        return 'Added backup custom periodicity support';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql('ALTER TABLE `database` ADD backup_task_periodicity VARCHAR(255) NOT NULL, ADD backup_task_periodicity_number INT NOT NULL, ADD backup_task_start_from DATETIME NOT NULL, ADD backup_task_next_iteration DATETIME NOT NULL');
+        $this->addSql('UPDATE `database` SET backup_task_periodicity = :periodicity;', ['periodicity' => BackupTaskPeriodicity::WEEK->value]);
+        $this->addSql('UPDATE `database` SET backup_task_periodicity_number = 1;');
+        $this->addSql('UPDATE `database` SET backup_task_start_from = :start_from;', ['start_from' => (new \DateTime())->format('Y-m-d')]);
+        $this->addSql('UPDATE `database` SET backup_task_next_iteration = :next_iteration;', ['next_iteration' => (new \DateTime('+1 week'))->format('Y-m-d')]);
```
Comment (jmsche) [inline migrations/Version20220801120509.php:24]: Can be done in one query instead

**Review by jmsche — APPROVED** [review]

File: migrations/Version20220801120509.php (hunk 4 above)
Comment (bastien70) [reply migrations/Version20220801120509.php:24]: I found it cleaner to do it like that. Is it really necessary to put it on one line?

Comment (jmsche) [reply migrations/Version20220801120509.php:24]: One query means it would run faster

Comment (bastien70) [reply migrations/Version20220801120509.php:24]: Something like this ?

```php
$this->addSql('ALTER TABLE `database` ADD backup_task_periodicity VARCHAR(255) NOT NULL DEFAULT :periodicity, ADD backup_task_periodicity_number INT NOT NULL DEFAULT 1, ADD backup_task_start_from DATETIME NOT NULL DEFAULT :start_from, ADD backup_task_next_iteration DATETIME NOT NULL DEFAULT :next_iteration', [
            'periodicity' => BackupTaskPeriodicity::WEEK->value,
            'start_from' => (new \DateTime())->format('Y-m-d'),
            'next_iteration' => (new \DateTime('+1 week'))->format('Y-m-d'),
        ]);
```

Comment (jmsche) [reply migrations/Version20220801120509.php]: No, just one update statement after the alter table

Comment (bastien70) [reply migrations/Version20220801120509.php]: Updated

**Review by jmsche — COMMENTED** [review]
Docs should be updated I guess

**Comment by bastien70** [comment]
> Docs should be updated I guess

Maybe in a new PR after merged the 3 others ? Since there are a lot of changes in the doc between the different PRs, you might as well do it in another one right after

**Comment by jmsche** [comment]
Sure

**Review by jmsche — APPROVED** [review]

**Pull request merged**